project/cmake/scripts/common/addon-helpers.cmake

```diff
@@ -103,22 +114,46 @@ macro (build_addon target prefix libs)
       # is changed within Visual Studio)
       string(REPLACE "$(Configuration)" "${CMAKE_BUILD_TYPE}" LIBRARY_LOCATION "${LIBRARY_LOCATION}")
 
-      # install the generated DLL file
-      INSTALL(PROGRAMS ${LIBRARY_LOCATION} DESTINATION ${target}
-              COMPONENT ${target}-${${prefix}_VERSION})
-
-      IF(CMAKE_BUILD_TYPE MATCHES Debug)
-        # for debug builds also install the PDB file
-        get_filename_component(LIBRARY_DIR ${LIBRARY_LOCATION} DIRECTORY)
-        INSTALL(FILES ${LIBRARY_DIR}/${target}.pdb DESTINATION ${target}
+      if(${prefix}_SOURCES)
+        # install the generated DLL file
+        INSTALL(PROGRAMS ${LIBRARY_LOCATION} DESTINATION ${target}
                 COMPONENT ${target}-${${prefix}_VERSION})
-      ENDIF()
+
+        IF(CMAKE_BUILD_TYPE MATCHES Debug)
+          # for debug builds also install the PDB file
+          get_filename_component(LIBRARY_DIR ${LIBRARY_LOCATION} DIRECTORY)
+          INSTALL(FILES ${LIBRARY_DIR}/${target}.pdb DESTINATION ${target}
+                  COMPONENT ${target}-${${prefix}_VERSION})
+        ENDIF()
+      endif()
+      if (${prefix}_CUSTOM_BINARY)
+        list(GET ${prefix}_CUSTOM_BINARY 0 FROM_BINARY)
+        list(GET ${prefix}_CUSTOM_BINARY 1 TO_BINARY)
+        install(FILES ${FROM_BINARY} DESTINATION ${target} RENAME ${TO_BINARY})
+      endif()
+      if(${prefix}_CUSTOM_DATA)
+        install(DIRECTORY ${${prefix}_CUSTOM_DATA} DESTINATION ${target}/resources)
+      endif()
     ELSE(WIN32)
       if(NOT CPACK_PACKAGE_DIRECTORY)
         set(CPACK_PACKAGE_DIRECTORY ${CMAKE_BINARY_DIR})
       endif()
-      INSTALL(TARGETS ${target} DESTINATION ${target}
-              COMPONENT ${target}-${${prefix}_VERSION})
+      if(${prefix}_SOURCES)
+        INSTALL(TARGETS ${target} DESTINATION ${target}
+                COMPONENT ${target}-${${prefix}_VERSION})
+      endif()
+      if (${prefix}_CUSTOM_BINARY)
+        list(GET ${prefix}_CUSTOM_BINARY 0 FROM_BINARY)
+        list(GET ${prefix}_CUSTOM_BINARY 1 TO_BINARY)
+        if(OS STREQUAL "android")
+          set(TO_BINARY "lib${TO_BINARY}")
```

Comment:
Not sure I understand. This is only to be used for the final addon binary
which will always be shared unless you plan to add linker steps to addon
loading in kodi.
17. mai 2016 15:14 skrev "Wolfgang Schupp" <notifications@github.com>:

> In project/cmake/scripts/common/addon-helpers.cmake
> <https://github.com/xbmc/xbmc/pull/9829#discussion_r63518520>:
>
> >      ELSE(WIN32)
> >        if(NOT CPACK_PACKAGE_DIRECTORY)
> >          set(CPACK_PACKAGE_DIRECTORY ${CMAKE_BINARY_DIR})
> >        endif()
> > -      INSTALL(TARGETS ${target} DESTINATION ${target}
> > -              COMPONENT ${target}-${${prefix}_VERSION})
> > +      if(${prefix}_SOURCES)
> > +        INSTALL(TARGETS ${target} DESTINATION ${target}
> > +                COMPONENT ${target}-${${prefix}_VERSION})
> > +      endif()
> > +      if (${prefix}_CUSTOM_BINARY)
> > +        list(GET ${prefix}_CUSTOM_BINARY 0 FROM_BINARY)
> > +        list(GET ${prefix}_CUSTOM_BINARY 1 TO_BINARY)
> > +        if(OS STREQUAL "android")
> > +          set(TO_BINARY "lib${TO_BINARY}")
>
> although its not likely to happen at the moment, but what about non-shared
> libs?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/xbmc/xbmc/pull/9829/files/76d4426cda2a5f44e6c561d7d46764b95b4004f9#r63518520>
>